File: Algorithms/309. Best Time to Buy and Sell Stock with Cooldown/golang/solution.go
func maxProfit(prices []int) int {
	profitOne := 0
	profitTwo := 0
	for i := 1; i < len(prices); i++ {
		tmp := profitOne
		profitOne = max(profitOne+prices[i]-prices[i-1], profitTwo)
		profitTwo = max(tmp, profitTwo)
	}
	return max(profitOne, profitTwo)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: Algorithms/309. Best Time to Buy and Sell Stock with Cooldown/javascript/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */

var maxProfit = function (prices) {
    let sell = 0;
    let buy = -prices[0];
    let cooldown = 0;

    for (let i = 1; i < prices.length; i++) {
        const nextSell = Math.max(sell, buy + prices[i]);
        const nextBuy = Math.max(buy, cooldown - prices[i]);
        const nextCooldown = Math.max(cooldown, sell);

        sell = nextSell;
        buy = nextBuy;
        cooldown = nextCooldown;
    }

    return sell;
};
